modules/network-annotation-processor/src/main/java/org/apache/ignite/internal/network/processor/TransferableObjectProcessor.java

```diff
@@ -351,4 +354,33 @@ private void writeServiceFile(String packageName, TypeSpec registryInitializer)
             throw new ProcessingException(e.getMessage(), e);
         }
     }
+
+    /** Generates utility classes for {@link Enum} in network messages. */
+    private void generateEnumTransferableUtils(Collection<MessageClass> messageClasses, MessageGroupWrapper messageGroup) {
+        List<MessageClass> serializableMessages = collectAutoSerializableMessageClasses(messageClasses);
+
+        if (serializableMessages.isEmpty()) {
+            return;
+        }
+
+        var enumMethodsGenerator = new EnumMethodsGenerator(processingEnv);
+
+        Set<TypeMirror> enumTypes = enumMethodsGenerator.collectEnums(messageClasses);
+
+        if (enumTypes.isEmpty()) {
+            return;
+        }
```

Comment:
This seems to be redundant: if the set is empty, the following code will do nothing

Reply:
Ok, why not.